File: Subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    if (s.length > t.length) return false;
    if (s.length === 0) return true;

    let i = 0, j = 0;

    while (i < t.length && j < s.length) {
        if (t[i] === s[j]) {
            j++;
        }
        i++;
    }

    return j === s.length;
}

// Example usage
console.log(isSubsequence("abc", "ahbgdc")); // true
console.log(isSubsequence("axc", "ahbgdc")); // false

File: Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) > len(t):
            return False
        if len(s) == 0:
            return True

        i, j = 0, 0
        while i < len(t) and j < len(s):
            if t[i] == s[j]:
                j += 1
            i += 1

        return j == len(s)

print(Solution().isSubsequence("abc", "ahbgdc"))  # True
print(Solution().isSubsequence("axc", "ahbgdc"))  # False
